crates/task-impls/src/quorum_vote/handlers.rs

```diff
@@ -287,7 +287,7 @@ pub(crate) async fn handle_quorum_proposal_validated<
             .last_decided_view
             .set(usize::try_from(consensus_writer.last_decided_view().u64()).unwrap());
         let cur_number_of_views_per_decide_event =
-            *task_state.latest_proposed_view - consensus_writer.last_decided_view().u64();
+            *proposal.view_number() - consensus_writer.last_decided_view().u64();
```

Comment:
I'm not sure this is a 1:1 change. The original code was comparing the most recently submitted proposal, but this one is pointing to the latest proposal that we've received, which can be a much different view number (i.e. in the case of timeouts or view sync delays). I don't know if it's wrong necessarily, but I just want to highlight that this is gonna be a different code path running.

Any thoughts @bfish713?

Reply:
I agree it's not a 1:1 change, but I think it aligns with the meaning of this metric, i.e., "Number of views that are in-flight since the last anchor view". Since we moved the proposal validation from the proposal task to the vote task, the latest view "in-flight" is just ` proposal.view_number()` for the vote task.

If we specifically want to use the latest proposed view (rather than the latest seen view) in the calculation, then yes, I think this metric update should be moved to the proposal task. But I still feel it's better to keep this in the vote task where we update the decided view.

Reply:
Yeah I  don't think I disagree with this, just wanted to call it out, I assume there's nothing downstream that depends on a certain pattern to the metrics.